### ISSUE TITLE
bccH 2x2x2 and 3x3x3 runs and derivative checks are failing

### ISSUE THREAD

**Issue by prckent** [QMCPACK/qmcpack#442]
```
Super twist #0:  [   0.00000   0.00000   0.00000 ]
Using supercell twist 0:  [   0.00000   0.00000   0.00000]
Using 1 copies of twist angle [-0.000, -0.000, -0.000]
Using real orbitals.
Fatal Error. Aborting at backflow optimization is broken with UseRealOrbitals
```
Lots of breakage on cdash. 

I suspect a simply incomplete merge or something wrong with the test definition. Hopefully an easy fix.

**Comment by Paul-St-Young** [comment]
@prckent, my current workarounds for these failures are to re-route code path. For example, to fix the above test, I simply force `UseRealOrbitals=true` at the current abort. After discussion with @ye-luo in #373, we have decided revert the workaround and leave the broken tests in the PR to alert problems.

@prckent and @ye-luo, please let me know how to proceed. It is not immediately clear to me where the bug is. The current tests will result in a red cdash. For a green cdash, I can either:
1. re-apply workarounds OR
2. remove broken tests

**Comment by ye-luo** [comment]
@Paul-St-Young I think your workaround force `UseRealOrbitals=true` If you did a typo in the previous post, please edit it and remove this comment.

**Comment by ye-luo** [comment]
In principle, if complex is good, fixing the real case is fairly close. Find the correspondent code path in EinsplineSet.cpp and compare it with the complex working codepath.

**Comment by ye-luo** [comment]
There are three code path Spline to SPO, R2R and C2R and C2C. The old EinsplineSet also has a similar setting.
R2R: HalfG is either 0 or 1 for distinguishing Gamma and X.
C2R: When you use tiling and your twist angle is 0 or 0.5, use exp(k,r) to apply the phase in the supercell. This k is not the twist angle but the k point of the primitive cell. MakeTwoCopies is because of symmetry. If true, each complex spline contributes two real SPO.
C2C: Any twist angle. Each complex spline always contributes a complex SPO.

**Comment by ye-luo** [comment]
resolved by #1204
